### ISSUE TITLE
Entry point should probably be a console script

### ISSUE THREAD

**Issue by hoechenberger** [psyfood/pyqmix-web#43]
Currently the entry point is specified as a gui script:
https://github.com/psyfood/pyqmix-web/blob/3a207b9da9169b784e9bfd53b3c81d3c90e15332/setup.cfg#L32

I believe we ran into issues with this in the past, and it should be a console script instead.

**Comment by hoechenberger** [comment]
@ArndalAndersen Could you check whether the entry point works on Widnows? i.e., if after installing the package, running `pyqmix-web` fromt he console fires up the application?
I believe we had to change `gui_scripts` to `console_scripts` in the past to make this work…

**Comment by ArndalAndersen** [comment]
@hoechenberger, I also remember this. Can check it again for sure.

**Comment by hoechenberger** [comment]
That would be great. Thanks.

**Comment by ArndalAndersen** [comment]
What I did:
1. I changed `gui_scripts` to `console_scripts` in `setup.cfg`
2. Packaged it via `python setup.py sdist bdist_wheel`
````
$ conda create -n testenv python=3
$ conda activate
$ pip install dist\pyqmix_web-2019.1+0.g3a207b9.dirty.tar.gz 
$ pyqmix-web
```

The error I get in the console:
```
Traceback (most recent call last):
  File "c:\users\xt6523\appdata\local\continuum\miniconda3\envs\test4\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\xt6523\appdata\local\continuum\miniconda3\envs\test4\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\xt6523\AppData\Local\Continuum\miniconda3\envs\test4\Scripts\pyqmix-web.exe\__main__.py", line 5, in <module>
  File "c:\users\xt6523\appdata\local\continuum\miniconda3\envs\test4\lib\site-packages\pyqmix_web\run.py", line 2, in <module>
    from pyqmix_backend.backend_app import app
ModuleNotFoundError: No module named 'pyqmix_backend'
```

**Comment by hoechenberger** [comment]
I believe (as in: haven't tried it out yet) it should work if that import in `run.py` is changed from
```
from pyqmix_backend.backend_app import app
```
to
```
from pyqmix.backend_app import app
```
This should run fine when installed for development (`pip install -e .`) and as a package.

**Comment by ArndalAndersen** [comment]
Sure looks like it's working now :) GO YOU!

I tested it on my mac though. You asked me to test it on a Windows computer, which I can do too, but I then started wondering why that's even necessary?

**Comment by hoechenberger** [comment]
> Sure looks like it's working now :) GO YOU!

Yay!

> You asked me to test it on a Windows computer, which I can do too, but I then started wondering why that's even necessary?

Because entry points are OS-dependent and could behave differently across platforms.

**Comment by ArndalAndersen** [comment]
Before I went ahead and closed this issue I wanted to check whether the _standalone_ would still work when I adapted the import setting in `run.py`:
`from pyqmix_backend.backend_app import app`
to
`from pyqmix_web.backend_app import app`

Unfortunately it doesn't. When I start the standalone it crashes because it cannot find the module named `pyqmix_web`.

**Comment by hoechenberger** [comment]
Darn! Did you install pyqmix (`pip install .`, _not_ a development install) before building the standalone package? Wonder whether this could be the cause… Otherwise, we'll just have to add a `try ... except` block around the imports...

**Comment by hoechenberger** [comment]
By the way, you can also build and run the standalone on macOS the same way as you do on Windows!

Just be aware that
- pyinstaller currently only works with Python 3.6 and
- requires ` jsonschema=2.6.0` (newer versions will not work and generate weird error message)

The executable will then be placed in `pyqmix_backend/dist/pyqmix-web`.

**Comment by hoechenberger** [comment]
Okay so I just tested _on macOS_, like I proposed above:
- properly install `pyqmix_web` via `pip install -U .` from the `pyqmix-web` root directory
- then run `pyinstaller --clean run.spec` from `pyqmix-web/pyqmix_backend`
- then run the generated executable: `./dist/pyqmix-web`

And it works!

**Comment by ArndalAndersen** [comment]
> Darn! Did you install pyqmix (pip install ., not a development install) before building the standalone package? Wonder whether this could be the cause… Otherwise, we'll just have to add a try ... except block around the imports, I'm afraid!

Nope. I did not install `pyqmix_web` as this is not part of our normal [routine](https://github.com/psyfood/pyqmix-web/blob/master/EXECUTABLE.md) to build a standalone. Actually, I am puzzled why it's necessary to install the package before creating its standalone - I have never needed to previously (btw. I assume you meant `pyqmix_web` and `pyqmix` in the above). I guess the only reason to install it would be such that the import statement actually works (`from pyqmix_web.backend_app import app`). 

> By the way, you can also build and run the standalone on macOS the same way as you do on Windows!
Just be aware that pyinstaller currently only works with Python 3.6 and
requires jsonschema=2.6.0 (newer versions will not work and generate weird error message)
The executable will then be placed in pyqmix_backend/dist/pyqmix-web.

Yep, that's actually what I've been doing all along for testing. You will see that I already updated `EXECUTABLE.md` with information about version numbers for `jsonschema`.  I don't recall that Python had to be in version 3.6 though.  

> Okay so I just tested on macOS, like I proposed above:
properly install pyqmix_web via pip install -U . from the pyqmix-web root directory
then run pyinstaller --clean run.spec from pyqmix-web/pyqmix_backend directory
then run the generated executable: ./dist/pyqmix-web

Yay! Again, I did not install pyqmix_web when I created the standalone, so that might definitely be the issue. But then we really should update our `EXECUTABLE.md` [recipe](https://github.com/psyfood/pyqmix-web/blob/master/EXECUTABLE.md) 

Will try it out today.

**Comment by hoechenberger** [comment]
> I have never needed to previously (btw. I assume you meant `pyqmix_web` and `pyqmix` in the above). I guess the only reason to install it would be such that the import statement actually works (`from pyqmix_web.backend_app import app`).

Yes exactly. I think it has to do with the way that `PyInstaller` checks which packages / dependencies to package. If it sees we require the package `pyqmix_web`, however this package was just installed via `pip install -e` (i.e., development mode), the packaging doesn't work correctly. So therefore my idea was to install `pyqmix_web` "properly" (without the `-e` flag), which seems to work. :)

> You will see that I already updated `EXECUTABLE.md` with information about version numbers for `jsonschema`. I don't recall that Python had to be in version 3.6 though.

Cool! I figured the Python 3.6 limitation is simply due to `conda-forge`: currently, `PyInstaller` is [not available](https://github.com/conda-forge/pyinstaller-feedstock/issues/32) for newer Python versions on macOS (but it _is_ available for Python 3.7 on Linux and Windows).

> Again, I did not install pyqmix_web when I created the standalone, so that might definitely be the issue. But then we really should update our `EXECUTABLE.md` [recipe](https://github.com/psyfood/pyqmix-web/blob/master/EXECUTABLE.md)

Yes, if everything works for you after installing `pyqmix_web`, we should update these directions accordingly! Thanks!

**Comment by ArndalAndersen** [comment]
I installed `pyqmix-web` and sure enough the standalone now worked - even on Windows. I have never previously had to install `pyqmix-web` to compile it into its standalone, and am still a bit puzzled why it's necessary. But hey, it works, and that's fantastic. I created a pull request which solves this issue.
